### DISCUSSION TITLE
A chart to show working and not working material...

### DISCUSSION THREAD

**Discussion by antonycartwright** [dotnet/maui#21685] [General]
Hi all!

I've spent a few hours every day for 2 weeks on Maui. I'd say I've got a fairly good grip on how it works, etc. I like many of the concepts of Maui.

However, I unfortunately feel as though that many of the "basic" features of Maui simply don't work. I'm finding that I'm spending hours on simple issues to find out that they're known to be unfinished controls/views.

Today for example, I came to use a grid to find that it can't shrink back once stretched. I noticed that rowspacing sometimes doesn't work. I noticed that you can't build any type of view over a table - you need to basically put a table in a header and another table in the item. The carousel is unfortunately terrible and probably shouldn't even be in the toolbox. I've spent 3hrs on it to find that it's well-known to not work. The problems with the carousel are basically that it acts crazy. It doesn't work with a grid inside. Without a grid, it's a little better but it won't swipe. You can use the keyboard to go left and right, but it acts crazy. It sort of flickers/jumps all over the place and seems to pick a random item to display. The stepper control is glitchy - the minus button seems to not register if you approach it from the right. I can go on and on.

As much as I love Maui, I can't use it professionally because I simply don't know which aspects of it that I can reliably pump my time into. I realise that it's a massive project, but why push large swathes of knowingly broken material to us? If Maui literally consists of bindings, layouts, entry and labels at the moment - so be it! Tell us that! Narrow your focus to getting those 100%. Remove the other controls for now and focus on getting just the basics spot on! When the basics are spot on, then add in a new control one by one. This way we know that whatever is there, however "small" it may look is known to work.

It seems to me that far too much has been taken on, and it's being marketed to be able to do x, y and z, but it can't!

At the moment it's frustrating because we don't know what actually works and what doesn't. I want to carry on with Maui, but before I spend more time on controls, I just want to know which work. :-(

Antony

**Comment by HuaFangYun** [comment]
I think for now we can only look forward to quick fixes and quick merging of submitted security fixes to speed up developers' expectations of Maui. For now, use it first. There is more feedback when there is a problem, and it is hoped that Maui developers can invest more people and time to solve basic problems quickly while optimizing known performance issues. It is my idea to change the way people think about maui when they encounter problems.
